GetStream/stream-chat-android
# [3872] Deprecate ChatLogger and replace it with StreamLog

### 🎯 Goal

Deprecate ChatLogger and replace it with StreamLog

### 🧪 Testing

- Apply the patch below
- Run any sample app
- Make sure no SDK logs can be seen in Logcat

<details>

<summary>Apply this patch before testing.</summary>

```
Index: stream-chat-android-client/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-client/build.gradle b/stream-chat-android-client/build.gradle
--- a/stream-chat-android-client/build.gradle	(revision 5134fb443b5db71b13d043a5f94438e0b67e3411)
+++ b/stream-chat-android-client/build.gradle	(date 1657718892018)
@@ -59,7 +59,7 @@
 dependencies {
     api project(':stream-chat-android-core')
     api project(":libraries:logging:stream-logger-android")
-    debugApi project(":libraries:logging:stream-logger-android-init")
+    //debugApi project(":libraries:logging:stream-logger-android-init")
 
     implementation Dependencies.kotlinReflect
 
Index: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt
--- a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt	(revision 5134fb443b5db71b13d043a5f94438e0b67e3411)
+++ b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt	(date 1657718892010)
@@ -68,7 +68,7 @@
         val client = ChatClient.Builder(apiKey, context)
             .loggerHandler(FirebaseLogger)
             .notifications(notificationConfig, notificationHandler)
-            .logLevel(logLevel)
+            .logLevel(ChatLogLevel.NOTHING)
             .withPlugin(offlinePlugin)
             .debugRequests(true)
             .build()
Index: stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatHelper.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatHelper.kt b/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatHelper.kt
--- a/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatHelper.kt	(revision 5134fb443b5db71b13d043a5f94438e0b67e3411)
+++ b/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatHelper.kt	(date 1657718892024)
@@ -65,7 +65,7 @@
         ChatClient.Builder(apiKey, context)
             .notifications(notificationConfig, notificationHandler)
             .withPlugin(offlinePlugin)
-            .logLevel(logLevel)
+            .logLevel(ChatLogLevel.NOTHING)
             .build()
     }
 
```

</details>


### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_
